File: src/modules/kargs_manager.rs
use colored::Colorize;
use std::process::Command;

fn run_command(command: &str, arg: &str, description: &str) {
    let full_command = format!("{} {}", command, arg);

    let status = Command::new("/bin/sh")
        .args(["-c", full_command.as_str()])
        .status()
        .expect("Failed to run the command");

    if !status.success() {
        eprintln!("{}: Failed to {}", "ERR".bold().red(), description);
    }
}

pub fn remove_argument(karg: String) {
    run_command(
        "rpm-ostree",
        format!("kargs --delete={}", karg.as_str()).as_str(),
        "remove a kernel argument",
    )
}

pub fn replace_value(karg: String, new_value: String) {
    run_command(
        "rpm-ostree",
        format!("kargs replace={}={}", karg.as_str(), new_value.as_str()).as_str(),
        "replace the kernel argument",
    );
}

pub fn add_argument(karg: String) {
    run_command(
        "rpm-ostree",
        format!("kargs --append={}", karg.as_str()).as_str(),
        "add a kernel argument",
    );
}

File: modules/kargsManager.go
package modules

import (
	"os/exec"

	"github.com/darkmatterorg/nebula/config"
	"github.com/darkmatterorg/orbit/utils"
)

func Add_karg(karg string) {
	if karg == "" && config.Config.Insults {
		utils.Error("Maybe add the kernel argument idiot?")
	} else if karg == "" {
		utils.Error("Please add the kernel argument")
	}

	add_argument := exec.Command("rpm-ostree", "kargs", "--append=", karg)

	err := add_argument.Run()

	if err != nil {
		utils.Error("Failed to add the kernel argument!")
	}
}

func Remove_karg(karg string) {
	if karg == "" && config.Config.Insults {
		utils.Error("Maybe add the kernel argument idiot?")
	} else if karg == "" {
		utils.Error("Please add the kernel argument")
	}

	remove_argument := exec.Command("rpm-ostree", "kargs", "--delete=", karg)

	err := remove_argument.Run()

	if err != nil {
		utils.Error("Failed to delete the kernel argument!")
	}
}

func Replace_karg(karg string, new_value string) {
	if karg == "" && config.Config.Insults {
		utils.Error("Maybe add the kernel argument idiot?")
	} else if karg == "" {
		utils.Error("Please add the kernel argument")
	}

	replace_argument := exec.Command("rpm-ostree", "kargs", "--replace=", karg, "=", new_value)

	err := replace_argument.Run()

	if err != nil {
		utils.Error("Failed to replace the kernel argument!")
	}
}
